webapp/channels/src/components/widgets/menu/menu_modals/submenu_modal/submenu_modal.tsx

```diff
@@ -17,63 +17,58 @@ type Props = {
     onExited: () => void;
 }
 
-type State = {
-    show: boolean;
-}
-
-export default class SubMenuModal extends React.PureComponent<Props, State> {
-    public constructor(props: Props) {
-        super(props);
-        this.state = {
-            show: true,
-        };
-    }
+const SubMenuModal = ({
+    elements,
+    onExited,
+}: Props) => {
+    const [show, setShow] = useState(true);
 
-    public onHide = () => { //public because it is used on tests
-        this.setState({show: false});
+    const onHide = () => {
+        setShow(false);
     };
 
-    public render() {
-        let SubMenuItems;
-        if (this.props.elements) {
-            SubMenuItems = this.props.elements.map((element) => {
-                return (
-                    <Menu.ItemSubMenu
-                        key={element.id}
-                        id={element.id}
-                        text={element.text}
-                        subMenu={element.subMenu}
-                        action={element.action}
-                        filter={element.filter}
-                        root={false}
-                    />
-                );
-            });
-        }
-        return (
-            <Modal
-                dialogClassName={'SubMenuModal a11y__modal mobile-sub-menu'}
-                show={this.state.show}
-                onHide={this.onHide}
-                onExited={this.props.onExited}
-                enforceFocus={false}
-                id='submenuModal'
-                role='dialog'
-            >
-                <Modal.Body
-                    onClick={this.onHide}
-                >
-                    <MenuWrapper>
-                        <Menu
-                            openLeft={true}
-                            ariaLabel={Utils.localizeMessage('post_info.submenu.mobile', 'mobile submenu').toLowerCase()}
-                        >
-                            {SubMenuItems}
-                        </Menu>
-                        <div/>
-                    </MenuWrapper>
-                </Modal.Body>
-            </Modal>
+    let SubMenuItems;
+    if (elements) {
+        SubMenuItems = elements.map(
+            (element) => (
+                <Menu.ItemSubMenu
+                    key={element.id}
+                    id={element.id}
+                    text={element.text}
+                    subMenu={element.subMenu}
+                    action={element.action}
+                    filter={element.filter}
+                    root={false}
+                />),
```

Comment:
Can we wrap this into a useMemo? Something like...
```suggestion
    const subMenuItems = useMemo(() => {
        if (!elements) {
            return undefined;
        }
        return elements.map(
            (element) => (
                <Menu.ItemSubMenu
                    key={element.id}
                    id={element.id}
                    text={element.text}
                    subMenu={element.subMenu}
                    action={element.action}
                    filter={element.filter}
                    root={false}
                />),
            );
        }, [elements]);
```

Also, SubMenuItems is not really a component, so let's change the name to lowercase (i.e. subMenuItems`)